# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webscheduling/WebTaskSchedulerWorker.cpp
# Commit: 41d90165c19a
# Full Hash: 41d90165c19adfa61486484b7ffb5ccab599461c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-04-01 09:40:15
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Allow selecting tasks from all existing web schedulers for main thread r=jjaschke,smaug
#   
#   This is a missing step in the current implementation. When we about
#   to select a task, it should pick one from all scheduled tasks, including
#   the ones scheduled by different schedulers.
# ==============================================================================

diff -r 2b4f7ebdb0b7 -r 41d90165c19a dom/webscheduling/WebTaskSchedulerWorker.cpp
--- a/dom/webscheduling/WebTaskSchedulerWorker.cpp	Mon Mar 31 19:49:02 2025 +0000
+++ b/dom/webscheduling/WebTaskSchedulerWorker.cpp	Mon Mar 31 19:49:03 2025 +0000
@@ -46,7 +46,8 @@
   aWorkerPrivate->AssertIsOnWorkerThread();
 
   if (mSchedulerWorker) {
-    RefPtr<WebTask> task = mSchedulerWorker->GetNextTask();
+    RefPtr<WebTask> task =
+        mSchedulerWorker->GetNextTask(false /* aIsMainThread */);
     if (task) {
       task->Run();
     }